src/app/shared/utils/utils.ts

```diff
@@ -218,4 +218,82 @@ export class Util {
   public static getFullName(person: Person): string {
     return `${person.lastName} ${person.firstName} ${person.middleName}`;
   }
+
+  /**
+   * Create filter state query string depending on current and default filter values
+   * @param filterState
+   * @return Query string
+   */
+  public static getFilterStateQuery(filterState): string {
+    let filterStateDiff: any = {};
+    let filterStateQuery = '';
+    const defaultFilterState = new DefaultFilterState();
+
+    // Compare current filter state and default
+    for (let [key, value] of Object.entries(defaultFilterState)) {
+      if (key === 'directionIds' || key === 'workingDays' || key === 'statuses') {
+        continue;
+      }
+      if (value !== filterState[key]) {
+        filterStateDiff[key] = filterState[key];
+      }
+    }
+
+    if (filterState.directionIds?.length > 0 ) {
+      filterStateDiff.directionIds = filterState.directionIds.join();
+    }
+
+    if (filterState.workingDays?.length > 0 ) {
+      filterStateDiff.workingDays = filterState.workingDays.join();
+    }
+
+    if (filterState.statuses?.length > 0 ) {
+      filterStateDiff.statuses = filterState.statuses.join();
+    }
+
+    // Create query string from filterStateDiff object
+    Object.keys(filterStateDiff).forEach((key,index, keyArray) => {
+      // Shouldn't add semicolon on last iteration
+      if (index === keyArray.length - 1) {
+        filterStateQuery += `${key}=${filterStateDiff[key]}`;
+      }
+      else {
+        filterStateQuery += `${key}=${filterStateDiff[key]};`;
+      }
+    });
+
+    return filterStateQuery;
+  }
+
+  /**
+   * Parse query string into filter state object to merge
+   * @param params - filter parameter in url query string
+   */
+  public static parseFilterStateQuery(params: string) {
```

Comment:
return typo

Reply:
typo?)

Reply:
https://tenor.com/view/confused-john-travolta-gif-25105508

Reply:
Typo is short for typographical error—a mistake made when typing something.